### ISSUE TITLE
Configure deploy on release on GH pages

### ISSUE THREAD

**Issue by surchs** [neurobagel/query-tool#41]
We want this tool to auto-deploy on release to GH pages. To do so:

- ~[ ] Make sure this repo has our auto-release workflow setup (with labels etc)~
- [x] Configure a GH publish workflow that only triggers for a new release
- [x] (optional) have staging and preview (main but no release yet) deploys continue to run on netlify

**Comment by rmanaem** [comment]
@surchs 
> (optional) have staging and preview (main but no release yet) deploys continue to run on netlify

Is this something we can do on Netlify or does it require configuration on GH?

**Comment by surchs** [comment]
Yeah, we could just have that. We already have the staging builds on netlify, so we could just use netlify as a preview of the current HEAD of the main branch, but only deploy to GH pages as the production server whenever we make a release. That way we could always look at the most recent version of the query tool in a public URL

**Comment by surchs** [comment]
@rmanaem I belive this is already the case btw: https://neurobagel-query.netlify.app/?node=All we could decide to switch to one of our domain names if we want that

**Comment by rmanaem** [comment]
Things learned while working on deployment CI:

From [vite docs](https://vitejs.dev/guide/static-deploy.html#github-pages):
> Set the correct base in vite.config.js.
If you are deploying to https://<USERNAME>.github.io/, or to a custom domain through GitHub Pages (eg. www.example.com), set base to '/'. Alternatively, you can remove base from the configuration, as it defaults to '/'. If you are deploying to https://<USERNAME>.github.io/<REPO>/ (eg. your repository is at https://github.com/{USERNAME}/{REPO}), then set base to '/{REPO}/'

If the event that base is set to something other than the default the router needs to be update to reflect the base URL as well so in our case `main.tsx` where we create a `browserRouter` would look something like this:

```tsx
import React from 'react';
import ReactDOM from 'react-dom/client';
import { createBrowserRouter, RouterProvider } from 'react-router-dom';
import { StyledEngineProvider } from '@mui/material/styles';
import App from './App';
import './index.css';

const baseURL = import.meta.env.BASE_URL || '/';

const router = createBrowserRouter([
  {
    path: baseURL,
    element: <App />,
  },
]);

ReactDOM.createRoot(document.getElementById('root')!).render(
  <React.StrictMode>
    {/* CSS injection order for MUI and tailwind: https://mui.com/material-ui/guides/interoperability/#tailwind-css */}
    <StyledEngineProvider injectFirst>
      <RouterProvider router={router} />
    </StyledEngineProvider>
  </React.StrictMode>
);
```

**Comment by surchs** [comment]
:rocket: Issue was released in [`v0.1.0`](https://github.com/neurobagel/query-tool/releases/tag/v0.1.0) :rocket:
